chatkit-core/src/main/kotlin/com/pusher/chatkit/cursors/Throttler.kt

```diff
@@ -0,0 +1,41 @@
+package com.pusher.chatkit.cursors
+
+import com.pusher.platform.network.Futures
+import com.pusher.platform.network.cancel
+import com.pusher.platform.network.map
+import com.pusher.platform.network.wait
+import com.pusher.util.Result
+import com.pusher.util.asFailure
+import elements.Error
+import elements.Errors
+import java.lang.Thread.*
+import java.util.concurrent.Future
+import kotlin.properties.Delegates
+
+/**
+ * Used to do a trailing throttle of actions. The [currentTask] is cancelled if a new action comes in before.
+ */
+internal class Throttler<A> {
+
+    private var currentTask by Delegates.observable<Task<A>?>(null) { _, old, _ ->
+        old?.cancel()
+    }
+
+    fun throttle(future: Future<Result<A, Error>>, delay: Long = 500): Future<Result<A, Error>> =
```

Comment:
My understanding of the behaviour described here is that once `setReadCursor` is called once (or anything that calls `throttler.throttle`) then a task is scheduled for 500ms into the future and if `setReadCursor` is called again within that 500ms window then the scheduled task is cancelled and a new one is scheduled for 500ms into the future again.

Is that correct?

If so, I think we want to change it so that once a call to `setReadCursor` is made then after 500ms a request will be made, but the payload of that request may change in that 500ms window if another call to `setReadCursor` is made (with a different position).

The JS implementation is probably the easiest place to grok the desired behaviour: https://github.com/pusher/chatkit-client-js/blob/3e2909fe568f17cb0430332b5bc357094f067f3b/src/current-user.js#L93-L115